src/TritonCTS/src/SinkClustering.cpp

```diff
@@ -167,6 +168,221 @@ void SinkClustering::findBestMatching()
   }
 }
 
+
+void SinkClustering::runAgglomerative(unsigned groupSize, float maxDiameter, cts::DBU scaleFactor)
```

Comment:
I believe this method could use a significant refactor. As a suggestion I think encapsulating a few of the loop bodies into helper functions would help readers get context on the intent of the code.